vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/BaseUIUnitTest.java

```diff
@@ -56,4 +61,90 @@ protected void cleanVaadinEnvironment() {
     protected String scanPackage() {
         return null;
     }
+
+    /**
+     * Navigate to the given view class if it is registered.
+     *
+     * @param navigationTarget
+     *         view class to navigate to
+     * @param <T>
+     *         view type
+     * @return instantiated view
+     */
+    public <T extends Component> T navigate(Class<T> navigationTarget) {
+        UI.getCurrent().navigate(navigationTarget);
+        return (T) getCurrentView();
```

Comment:
What about using `navigationTarget.cast(getCurrentView())`, so IntelliJ IDEA stops to complain about unchecked cast?
```suggestion
        return navigationTarget.cast(getCurrentView());
```